applications/nrf_desktop/src/util/hid_reportq.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <zephyr/sys/slist.h>
+#include <zephyr/kernel.h>
+
+#include "hid_reportq.h"
+#include "hid_report_desc.h"
+#include "hid_event.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(hid_reportq, CONFIG_DESKTOP_HID_REPORTQ_LOG_LEVEL);
+
+#define MAX_ENQUEUED_REPORTS CONFIG_DESKTOP_HID_REPORTQ_MAX_ENQUEUED_REPORTS
+
+struct enqueued_report {
+	sys_snode_t node;
+	struct hid_report_event *event;
+};
+
+struct counted_list {
+	sys_slist_t list;
+	size_t node_count;
+};
+
+struct hid_reportq {
+	struct counted_list reports[ARRAY_SIZE(input_reports)];
```

Comment:
?

```suggestion
	struct counted_list report_cnt_lists[ARRAY_SIZE(input_reports)];
```

Reply:
Renamed to `report_lists` (we may desist from using counted list in favor of simple linked list here)